contrib/analysis/merge_sos_and_perf_parallel.py

```diff
@@ -329,44 +331,39 @@ def transform_result(source, pbench_runs, results_seen, stats):
     result["controller_dir"] = pbench_run["controller_dir"]
     result["sosreports"] = pbench_run["sosreports"]
 
-    # optional workload parameters
-    try:
-        result["benchmark.filename"] = ", ".join(
-            set((benchmark["filename"].split(",")))
-        )
-    except KeyError:
-        result["benchmark.filename"] = "/tmp/fio"
-    try:
-        result["benchmark.iodepth"] = benchmark["iodepth"]
-    except KeyError:
-        result["benchmark.iodepth"] = "32"
-    try:
-        result["benchmark.size"] = ", ".join(set((benchmark["size"].split(","))))
-    except KeyError:
-        result["benchmark.size"] = "4096M"
-    try:
-        result["benchmark.numjobs"] = ", ".join(set((benchmark["numjobs"].split(","))))
-    except KeyError:
-        result["benchmark.numjobs"] = "1"
-    try:
-        result["benchmark.ramp_time"] = benchmark["ramp_time"]
-    except KeyError:
-        result["benchmark.ramp_time"] = "none"
-    try:
-        result["benchmark.runtime"] = benchmark["runtime"]
-    except KeyError:
-        result["benchmark.runtime"] = "none"
-    try:
-        result["benchmark.sync"] = benchmark["sync"]
-    except KeyError:
-        result["benchmark.sync"] = "none"
-    try:
-        result["benchmark.time_based"] = benchmark["time_based"]
-    except KeyError:
-        result["benchmark.time_based"] = "none"
+    # optional workload parameters and their defaults
+    optional_workload_params_defaults = {
+        "filename" : "/tmp/fio",
+        "iodepth" : "32",
+        "size" : "4096M",
+        "numjobs" : "1",
+        "ramp_time" : "none",
+        "runtime" : "none",
+        "sync" : "none",
+        "time_based" : "none"
+    }
+
+    params_to_setify = {"filename": 1, "size": 1, "numjobs": 1}
+
+    for opt_param in optional_workload_params_defaults.keys():
+        try:
+            if opt_param in params_to_setify:
+                result[f"benchmark.{opt_param}"] = sentence_setify(benchmark[opt_param])
+            else:
+                result[f"benchmark.{opt_param}"] = benchmark[opt_param]
+        except KeyError:
+            result[f"benchmark.{opt_param}"] = optional_workload_params_defaults[opt_param]
 
     return result
```

Comment:
I agree that this code could use some tightening, but I'm not crazy about using a loop for that, particularly with the extra conditional inside it.  How about this, instead:
```
    result["benchmark.filename"] = sentence_setify(benchmark.get("filename", "/tmp/fio"))
    result["benchmark.iodepth"] = benchmark.get("iodepth", "32")
    result["benchmark.numjobs"] = sentence_setify(benchmark.get("numjobs", "1"))
    result["benchmark.ramp_time"] = benchmark.get("ramp_time", "none")
    result["benchmark.runtime"] = benchmark.get("runtime", "none")
    result["benchmark.size"] = sentence_setify(benchmark.get("size", "4096M"))
    result["benchmark.sync"] = benchmark.get("sync", "none")
    result["benchmark.time_based"] = benchmark.get("time_based", "none")
```
You could probably do better that that where the calls to `sentence_setify()` are made on default values, but you would have to make the code more complex.

Reply:
I initially thought that the code looked like it was being copy pasted with the same format but just the workload param changing, so figured a loop would be better to condense it. But I see now, that trying to do a loop or probably anything else fancier forces the use of more conditionals which are probably unnecessary and consume more time. I made the changes you suggested